### DISCUSSION TITLE
Detect when all components are initialized?

### DISCUSSION THREAD

**Discussion by rehhouari** [alpinejs/alpine#1257] [1. Help] [answered]
Is there a way to run code after Alpine.js initialized all components?
I'm writing a plugin that uses `Alpine.onComponentInitialized` to get data from specific components then that data is saved in an array.
The plugin need to call a method that will use that array.
However in the current form when the method is called the array isn't populated yet so the method doesn't do it's job;
```js
{
  arr: [],
  start() {
    Alpine.onComponentInitialized((component) => {
      // analyze and add data to this.arr
    }
    // call method that use this.arr
   this.run()
  },
  run() {
    console.log(this.arr)
    // this.arr is empty since the components arent initialized yet
  }
}
```

**Comment by rehhouari** [comment]
[here is the source](https://github.com/rehhouari/alpine-router/blob/0.0.1/src/index.js#L75) of the plugin if the example isn't clear enough

**Reply by SimoTod** [reply]
Depending on what you need (link does not work), you can try
`window.addEventListener('load', (event) => {...});`

**Reply by rehhouari** [reply]
Ah sorry about that forgot to make it public, it is now.
I've tried that but with same results 😕

**Reply by SimoTod** [reply]
I've only had a look from my mobile but shouldn't the first navigate (line 75) go inside componentInitialised after you have built all the routes?

**Reply by rehhouari** [reply]
Yes it does! It's the only way to be executed after they have initialized. Thank you ^^

**Comment by KevinBatdorf** [comment, marked as answer]
> to get data from specific components then that data is saved in an array.

Could you check the length of this array matches the length of components, and keep polling until it's ready?

**Reply by rehhouari** [reply]
Seems like it didn't send the reply, well I said Thank you it solved it, this works and allows having multiple routers ^^
